### ISSUE TITLE
Brainbrowser 2.4.2: options -double in minctoraw and -float64 in datatype

### ISSUE THREAD

**Issue by MounaSafiHarab** [aces/brainbrowser#303]
Loris integration of BrainBrowser sends the raw information (minctoraw) with the option -double and the headers with datatype -float64. This was working for us with brainbrowser 2.4.0. With brainbroswer 2.4.2, we now get the error:"RangeError: byte length of Float64Array should be a multiple of 8"
(we bypassed brainbrowser 2.4.1).
(Natacha, I can not seem to be able to assign to you, as per your request)

**Comment by rdvincent** [comment]
This is an alignment or integration issue. Changes from 2.4.0 to 2.4.2 are unlikely to account for this bug, so it may be a latent issue that is just being hit because of a particular file. It would be good to get a specific test case we can use to reproduce this.

**Comment by rdvincent** [comment]
@MounaSafiHarab Hi - is there a way that I can reproduce this bug in a development environment?

**Comment by natacha-beck** [comment]
@MounaSafiHarab Have you try to convert the file and use the BrainBrowser website ?

**Comment by MounaSafiHarab** [comment]
@rdvincent @natacha-beck;

Bert, it could be an integration issue on our end. The error is indeed happening with one file (the one that I happened to be testing with!, while other files seem to be fine). The file that now gives this error though did not give this error with brainbrowser 2.4.0, and using our Loris-Brainbrowser integration. We are essentially using minctoraw with -double option to generate the raw file, and we send the header information by specifying the datatype to be 'float64'. 

I will do what Natacha suggested later today as a first step, and get back to you!

**Comment by rdvincent** [comment]
@MounaSafiHarab OK, keep us posted. Even if there is a problem at your end, it would be useful for me to be able to examine the bug in case I can provide a fix. So if you do create the .header and .raw files for the .mnc file that is giving you trouble, I would be happy to give it a try myself and try to figure out exactly why the volume viewer is rejecting this file.

**Comment by MounaSafiHarab** [comment]
@natacha-beck:
as I do: 
$ nodejs minc2volume-viewer.js filename.mnc -T float64
the .raw and .header files load without any problem in BrainBRowser 2.4.2 on your website.

@rdvincent ,
When integrated within Loris, we do not generate the .raw and .header, but pass through the information to brainbrowser using the script in: 
https://github.com/aces/Loris/blob/16.1-dev/modules/brainbrowser/ajax/minc.php
Lines 33 and 76 are the problematic ones. Note that if I change line 33 to be minctoraw -byte -unsigned and line 76 to either be removed altogether or replaced with 'datatype'=>'uint8', then I get NO errors.

**Comment by rdvincent** [comment]
@MounaSafiHarab It's very possible that minctoraw is writing extra bytes onto the file - that could cause the message you are seeing. Does this happen with all files, or just certain files? Can you provide a file that exhibits the problem?

**Comment by MounaSafiHarab** [comment]
@rdvincent 

I have the problem with many files. 
I just sent you an example minc file by e-mail! Thanks a lot!

**Comment by MounaSafiHarab** [comment]
@natacha-beck @rdvincent 
The problem was related to web browser templates files that are being cached and which Loris was reusing. Others do not have this issue and when I rebooted and removed cached files,  I no longer see the error. 
I confirmed with Bert that using the command line, the raw files are exactly the correct size they are supposed to be... so it is not a brainbrowser 2.4.2 issue. Thanks!
